Stratis.Bitcoin/Features/MemoryPool/MemPoolCoinView.cs

```diff
@@ -9,35 +9,103 @@
 
 namespace Stratis.Bitcoin.Features.MemoryPool
 {
+    /// <summary>
+    /// Memory pool coin view.
+    /// Provides coin view representation of memory pool transactions via a backed coin view.
+    /// </summary>
     public class MempoolCoinView : CoinView , IBackedCoinView
     {
+        #region Fields
+
+        /// <summary>Transaction memory pool for managing transactions in the memory pool.</summary>
         private readonly TxMempool memPool;
-        private readonly AsyncLock mempoolScheduler;
+
+        /// <summary>A lock for managing asynchronous access to memory pool.</summary>
+        private readonly AsyncLock mempoolLock;
```

Comment:
Here I would very much like what I suggest in here - https://github.com/Aprogiena/StratisBitcoinFullNode/blob/d2e959613b87221ad7e5021f6a410bf39069181e/Documentation/commenting-policy.md#contracts

I.e. what you have here is a contract between mempoolLock and memPool and you should describe this contract. I.e. you would like to make sure that everyone who is using memPool is using the lock because it has to be protected.

You can check it here https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/Stratis.Bitcoin/BlockPulling/BlockPuller.cs#L84 how I use it - check the comment for the lockObject I have there and then e.g. for assignedBlockTasks. What I am looking for here is the remarks section as you can see in assignedBlockTasks - that tells the developers that they have to use the lock for all access. Somewhere else I have some other lock and some other object and I require the lock to be used for write access only, so then I just say that for write access you need to use the lock. So it is important for me to have the contract described properly.